api/v1alpha1/clusterconfig_types.go

```diff
@@ -186,6 +191,40 @@ func (ExtraAPIServerCertSANs) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+// ImageRegistryCredentials required for providing credentials for an image registry URL.
+type ImageRegistryCredentials struct {
+	// Registry URL.
+	URL string `json:"url"`
```

Comment:
I'm not sure URL is quite right here as technically this could be any `matchImages` glob couldn't it?